sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientTests.cs

```diff
@@ -700,113 +674,1998 @@ public async Task IsRunningReturnsTrueWhileStopProcessingAsyncIsNotCalled()
+            // after a shorter period of time.
+
             using var cancellationSource = new CancellationTokenSource();
-            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(15));
 
-            Assert.That(mockProcessor.Object.IsRunning, Is.False);
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
+
+            Assert.That(mockProcessor.Object.IsRunning, Is.False, "The failed processor should not be running anymore.");
 
             await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
 
-            Assert.That(mockProcessor.Object.IsRunning, Is.True);
+            Assert.That(mockProcessor.Object.IsRunning, Is.False, "StartProcessingAsync should not be able to reset processor state.");
+
+            try
+            {
+                await mockProcessor.Object.StopProcessingAsync(cancellationSource.Token);
+            }
+            catch (Exception) { }
+
+            await mockProcessor.Object.StartProcessingAsync(cancellationSource.Token);
+
+            Assert.That(mockProcessor.Object.IsRunning, Is.True, "The processor state should have been reset by StopProcessingAsync.");
 
             await mockProcessor.Object.StopProcessingAsync(cancellationSource.Token);
 
-            Assert.That(mockProcessor.Object.IsRunning, Is.False);
+            Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The processor should have stopped without cancellation.");
         }
 
         /// <summary>
-        ///   Verifies functionality of the <see cref="EventProcessorClient" /> properties.
+        ///   Verifies functionality of the <see cref="EventProcessorClient.StopProcessingAsync" />
+        ///   method.
         /// </summary>
         ///
         [Test]
-        public async Task IsRunningReturnsFalseWhenLoadBalancingTaskFails()
+        public async Task StopProcessingAsyncDoesNothingWhenProcessorIsNotRunning()
         {
-            var mockProcessor = new Mock<EventProcessorClient>(Mock.Of<StorageManager>(), "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), default, default) { CallBase = true };
-            var completionSource = new TaskCompletionSource<bool>();
+            var mockStorage = new Mock<MockCheckPointStorage>(default(Action<string>)) { CallBase = true };
+            var mockConsumer = new Mock<EventHubConsumerClient>("consumerGroup", Mock.Of<EventHubConnection>(), default);
+            var mockProcessor = new InjectableEventSourceProcessorMock(mockStorage.Object, "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), default, mockConsumer.Object);
 
-            // This should be called right before the first load balancing cycle.
+            // Ownership claim attempts with an empty owner identifier are actually relinquish requests. This should happen once
```

Comment:
Is this comment still accurate?   I thought that there was a dedicated method for this now.

Reply:
We have a dedicated method in the `PartitionLoadBalancer` class, but it basically makes a call to the storage's `ClaimOwnershipAsync` method. We are overriding the storage here.